modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -691,4 +695,18 @@ public static String sanitizeUri(String uri) {
         }
         return "/".concat(uri);
     }
+
+    public static void methodInvocationCheck(BStruct bStruct) {
+        if (bStruct.getNativeData(METHOD_ACCESSED) != null) {
+            throw new BallerinaException("illegal function invocation");
```

Comment:
I guess saying illegal state is better. Shall we check with java.

Reply:
IllegalStateException is used when a method has been invoked at an illegal or inappropriate time. So it is applicable. Made the change